Confus/Confus/Audio/PlayerAudioEmitter.cpp

```diff
@@ -1,98 +1,87 @@
+			: m_AttachedPlayer(a_AttachedPlayer)
         {
-        createAudioSources();
+			createAudioSources(a_AudioManager);
         }
 
         PlayerAudioEmitter::~PlayerAudioEmitter()
         {
-            for(auto audioSource : m_AudioSourceFootsteps)
-            {
-                delete(audioSource);
-            }
-            for(auto audioSource : m_AudioSourceGrunts)
-            {
-                delete(audioSource);
-            }
-            for(auto audioSource : m_AudioSourceSwordSwoshes)
-            {
-                delete(audioSource);
-            }
         }
 
-        void PlayerAudioEmitter::playFootStepSound() const
+        void PlayerAudioEmitter::playFootStepSound()
         {
-            if(!m_AudioSourceFootsteps[0]->isPlaying())
+            if(!m_AudioSourceFootsteps[0].isPlaying())
             {
-                m_AudioSourceFootsteps[0]->play();
+                m_AudioSourceFootsteps[0].play();
             }
-            else if(!m_AudioSourceFootsteps[1]->isPlaying())
+            else if(!m_AudioSourceFootsteps[1].isPlaying())
             {
-                m_AudioSourceFootsteps[1]->play();
+                m_AudioSourceFootsteps[1].play();
             }
             else
             {
-                m_AudioSourceFootsteps[2]->play();
+                m_AudioSourceFootsteps[2].play();
             }
         }
 
-
-        void PlayerAudioEmitter::playAttackSound(bool a_HeavyAttack) const
+        void PlayerAudioEmitter::playAttackSound(bool a_HeavyAttack)
         {
             if(!a_HeavyAttack)
             {
                 playRandomGrunt();
             }
             else
             {
-                m_AudioSourceGrunts[2]->play();
+                m_AudioSourceGrunts[2].play();
             }
             playRandomSwordSwosh();
         }
 
-        void PlayerAudioEmitter::playRandomGrunt() const
+        void PlayerAudioEmitter::playRandomGrunt()
         {
             std::srand(static_cast<int>(time(NULL)));
             auto randomNumber = std::rand() % 2;
 
             switch(randomNumber)
             {
             case 0:
-                m_AudioSourceGrunts[0]->play();
+                m_AudioSourceGrunts[0].play();
                 break;
             case 1:
-                m_AudioSourceGrunts[1]->play();
+                m_AudioSourceGrunts[1].play();
                 break;
             default:
                 break;
             }
         }
 
-        void PlayerAudioEmitter::playRandomSwordSwosh() const
+        void PlayerAudioEmitter::playRandomSwordSwosh()
         {
             std::srand(static_cast<int>(time(NULL)));
             auto randomNumber = std::rand() % 4;
 
             switch(randomNumber)
             {
             case 0:
-                m_AudioSourceSwordSwoshes[0]->play();
+                m_AudioSourceSwordSwoshes[0].play();
```

Comment:
Couldn't you just do m_AudioSourceSwordSwoshes[randomNumber].play() instead of the switch statement? (perhaps a Math.Floor() or Math.Ceil() if it is not a round number.)